### PR TITLE
Fix Player.setPlayerListOrder to send update to clients

### LINKED ISSUE #11727: Player Tablist Order API not working as expected.
### Expected behavior

Custom behaviour for the tablist order API.

### Observed/Actual behavior

Sorted alphabetically by player name no matter what.

### Steps/models to reproduce

Add LuckPerms and hook into it, make a weight-based tablist order:
```java
Bukkit.getScheduler().runTaskTimer(() -> {
for (final Player player : Bukkit.getOnlinePlayers()) {
    player.setPlayerListOrder(getPlayerWeight(player));
}
}, 0, 20L);
```
```java
private static int getPlayerWeight(Player player) {
    return RankUtils.getWeight(RankUtils.getLPRankName(player));
}
```
```java
public static int getWeight(String rank) {
    Group group = TurtledDuels.getLuckPerms().getGroupManager().getGroup(rank);

    if (group != null) {
        Collection<WeightNode> Nodes = group.getNodes(NodeType.WEIGHT);
        for (WeightNode node : Nodes) {
            return node.getWeight();
        }
    }

    return 0;
}

public static String getLPRankName(Player player) {
    String rankName = null;
    try {
        User user =  TurtledDuels.getLuckPerms().getPlayerAdapter(Player.class).getUser(player);
        rankName = user.getCachedData().getMetaData().getPrimaryGroup();
    } catch (Exception e) {
        return "default";
    }
    return rankName;
}
```

### Plugin and Datapack List

[11:11:31 INFO]: Paper Plugins:
[11:11:31 INFO]:  - FancyHolograms, FancyNpcs, SlimeWorldPlugin
[11:11:31 INFO]: Bukkit Plugins:
[11:11:31 INFO]:  - AxiomPaper, ContinuumWorlds, FastAsyncWorldEdit, LiteBans, LuckPerms, MiniMOTD, PlaceholderAPI, TCPShield, Tebex, TurtledDuels
[11:11:31 INFO]:  ViaBackwards, ViaVersion

### Paper version

```
[11:10:35 INFO]: Checking version, please wait...
[11:10:35 INFO]: This server is running AdvancedSlimePaper version 1.21.3-16488-HEAD@fbba5f8 (2024-12-05T11:33:21Z) (Implementing API version 1.21.3-R0.1-SNAPSHOT)
Error obtaining version information
Previous version: 1.21.1-DEV-a3d81f8 (MC: 1.21.1)
```
No datapacks

### Other

_No response_

### PR DESCRIPTION
**Pull request by 96DarkCode96**
fix #11727 , sends PlayerInfoUpdatePacket to all players that can see modified player

### CODE REVIEW

File: patches/server/1076-Fix-Player.setPlayerListOrder-to-send-update-to-clie.patch [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+index 20f64850cb42d986358f01ffcdb42187e1684bca..b60f547c8616e9d4a858e79d036712f311152aea 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+@@ -606,6 +606,12 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
+         Preconditions.checkArgument(order >= 0, "order cannot be negative");
+ 
+         this.getHandle().listOrder = order;
++        if (getHandle().connection == null) return; // Updates are possible before the player has fully joined
++        for (ServerPlayer player : server.getHandle().players) {
++            if (player.getBukkitEntity().canSee(this)) {
```
Comment (electronicboy) [inline patches/server/1076-Fix-Player.setPlayerListOrder-to-send-update-to-clie.patch]: This also needs to consider the isListed API

**Review by electronicboy — CHANGES_REQUESTED** [review]

File: patches/server/1076-Fix-Player.setPlayerListOrder-to-send-update-to-clie.patch (hunk 1 above)
Comment (96DarkCode96) [reply patches/server/1076-Fix-Player.setPlayerListOrder-to-send-update-to-clie.patch]: should be updated

Comment (kennytv) [reply patches/server/1076-Fix-Player.setPlayerListOrder-to-send-update-to-clie.patch]: Does it? Being unlisted doesn't mean there's no player data. When re-listed, it only updates the listed status, nothing else

Comment (electronicboy) [reply patches/server/1076-Fix-Player.setPlayerListOrder-to-send-update-to-clie.patch]: Yea, no, I totally brainfarted, I misread some stuff, I was wrong here

Comment (96DarkCode96) [reply patches/server/1076-Fix-Player.setPlayerListOrder-to-send-update-to-clie.patch]: Personally I haven't used isListed API ever in my codes, so really don't know what it does in background, but as I went through, this packet is always send regardless of that isListed API.

Comment (electronicboy) [reply patches/server/1076-Fix-Player.setPlayerListOrder-to-send-update-to-clie.patch]: yea, I was wrong to say that this should consider the isListed API, that last commit needs reverting

Comment (96DarkCode96) [reply patches/server/1076-Fix-Player.setPlayerListOrder-to-send-update-to-clie.patch]: i'll try to revert that, i'm really fighting with PR system (first PR that I've ever done). But I need to learn that.

Comment (electronicboy) [reply patches/server/1076-Fix-Player.setPlayerListOrder-to-send-update-to-clie.patch]: You should just be able to `git revert 9c855b2` and then push

Comment (96DarkCode96) [reply patches/server/1076-Fix-Player.setPlayerListOrder-to-send-update-to-clie.patch]: okay, it should be reverted (forgot to sign it - used other device)

**Review by electronicboy — APPROVED** [review]

**Review by kennytv — APPROVED** [review]

**Pull request merged**